backend/supplements/getPatchNotesDetails_VAL.js

```diff
@@ -17,98 +17,80 @@ const getPatchNotesDetails_VAL = async (url, version) => {
         });
         const $ = cheerio.load(response);
 
+        $('head, meta, .metadata, .header').remove();
+        $('[data-testid="article-card-carousel"]').remove();
+
         const patchDetails = {
             agentUpdates: [],
             mapUpdates: [],
             bugFixes: []
         };
 
-        const roleSections = {
-            "AGENT UPDATES": patchDetails.agentUpdates,
-            "MAP UPDATES": patchDetails.mapUpdates,
-            "BUG FIXES": patchDetails.bugFixes
+        let currentSection = null;
+        let contentBuffer = [];
+        let inBugFixes = false;
+
+        const flushContentBuffer = () => {
+            if (currentSection && contentBuffer.length > 0) {
+                patchDetails[currentSection].push(contentBuffer.join(' '));
+                contentBuffer = [];
+            }
         };
 
-        $('.sc-4225abdc-0').each((index, element) => {
-            const sectionTitle = $(element).find('h1, h2').first().text().trim().toUpperCase();
-            if (roleSections[sectionTitle]) {
-                const content = $(element).html().trim();
-                if (content) {
-                    roleSections[sectionTitle].push(content);
-                }
+        $('*').each((index, element) => {
+            const text = $(element).text().trim().toUpperCase();
+
+            if (text.includes('AGENT UPDATES')) {
+                flushContentBuffer();
+                currentSection = 'agentUpdates';
+                inBugFixes = false;
+            } else if (text.includes('MAP UPDATES')) {
+                flushContentBuffer();
+                currentSection = 'mapUpdates';
+                inBugFixes = false;
+            } else if (text.includes('BUG FIXES')) {
+                flushContentBuffer();
+                currentSection = 'bugFixes';
+                inBugFixes = true;
+            } else if (inBugFixes && (text.includes('KNOWN ISSUES') || text.includes('RELATED ARTICLES'))) {
+                flushContentBuffer();
+                currentSection = null;
+                inBugFixes = false;
+                if (text.includes('RELATED ARTICLES')) return false;
+            } else if (!inBugFixes && text.includes('CONSOLE')) {
+                flushContentBuffer();
+                currentSection = null;
+            } else if (currentSection && $(element).is('p')) {
+                contentBuffer.push($.html(element));
```

Comment:
hm would be nice to avoid having any html in the final structure, in case the source html has something dangerous... wdyt about converting to a lossy intermediary format like Markdown? could use a library like `turndown` https://github.com/mixmark-io/turndown